src/coreclr/jit/codegenxarch.cpp

```diff
@@ -2755,9 +2755,80 @@ void CodeGen::genCodeForInitBlkUnroll(GenTreeBlk* node)
 #endif
         }
 
-        instruction simdMov = simdUnalignedMovIns();
-        for (unsigned regSize = XMM_REGSIZE_BYTES; size >= regSize; size -= regSize, dstOffset += regSize)
+        ClassLayout* layout = node->GetLayout();
+
+        auto findNextGCPtr = [layout](unsigned startOffset, unsigned* pNextGCPtrOffset) {
+            bool found = false;
+
+            for (unsigned slot = startOffset / TARGET_POINTER_SIZE; slot < layout->GetSlotCount(); slot++)
+            {
+                if (layout->IsGCPtr(slot))
+                {
+                    *pNextGCPtrOffset = slot * TARGET_POINTER_SIZE;
+                    found             = true;
+                }
+            }
+
+            return found;
+        };
```

Comment:
This always returns the last GC pointer, unless I'm missing something.

Reply:
you are right, missed `break`

Reply:
Fixed the issue and validated on a struct with 6 GCPtr fields that they are zeroed with 6 `movq`-s